airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumStateDecoratingIterator.java

```diff
@@ -156,9 +156,12 @@ protected AirbyteMessage computeNext() {
 
         if (checkpointOffsetToSend.size() == 1
             && changeEventIterator.hasNext()
-            && !event.isSnapshotEvent()
-            && targetPosition.isEventAheadOffset(checkpointOffsetToSend, event)) {
-          sendCheckpointMessage = true;
+            && !event.isSnapshotEvent()) {
+          if (targetPosition.isEventAheadOffset(checkpointOffsetToSend, event)) {
+            sendCheckpointMessage = true;
+          } else {
+            LOGGER.info("Encountered {} records with the same event offset", recordsLastSync);
```

Comment:
This log message is causing our airbyte logs to significantly grow in size as we commonly see this message >1 million times in each sync. 

Is it possible to selectively disable this log or would we have to change our log level? We don't believe multiple records with the same offset is an issue for us because most of our database operations are batch inserts.

Reply:
@WillKoehrsen are you able to submit a PR which limits how often this is logged? Maybe only once per offset or stop after 100 logs?

Reply:
Sorry, I don't have the experience with the codebase to feel comfortable opening a PR. We were able to get around the issue in the short term by downgrading our postgres connector to 3.3.12.

Reply:
ok, we'll track this bug via https://github.com/airbytehq/airbyte/issues/36903

Reply:
@WillKoehrsen we have removed this log line in the new version of source-postgres

Reply:
Thanks a lot! Really appreciate it.